fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialogBottomSheet.kt

```diff
@@ -0,0 +1,234 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Divider
+import org.mozilla.fenix.compose.SwitchWithLabel
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Firefox Translation options bottom sheet dialog.
+ */
+@Composable
+fun TranslationOptionsDialogBottomSheet(
+    translationOptionsList: List<TranslationSwitchItem>,
+    onBackClicked: () -> Unit,
+    onTranslationSettingsClicked: () -> Unit,
+    aboutTranslationClicked: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer2,
+                shape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp),
+            ),
+    ) {
+        TranslationOptionsDialogHeader(onBackClicked)
+
+        LazyColumn {
+            items(translationOptionsList) { item: TranslationSwitchItem ->
+                SwitchWithLabel(
+                    checked = item.isEnabled,
+                    onCheckedChange = item.onStateChange,
+                    label = item.textLabel,
+                    modifier = Modifier
+                        .padding(start = 72.dp, end = 16.dp)
+                        .clearAndSetSemantics {
```

Comment:
The labels for the toggles aren't associated with TalkBack - I think this might be causing it, not 100% sure and didn't look in-depth. 

When testing, and you go to the toggle using TalkBack, it should:
* Have a label that is programatically connected to it - `name`
* Know the state of the toggle - `value` - inherited from `Switch` (please don't override the content description)
* Know it is a toggle - `role` - inherited from `Switch`

For an example, check out the `Menu` -> `Accessibility` -> and try the toggle for `Automatic font sizing`.

Notice, you can also double tap the label and it will toggle the switch. It shows they were programmatically associated.

When testing this component, it should behave similarly.

Reply:
I think I fixed it . There were some bugs in the SwitchWithLabel component.

Reply:
(Please see other conversation about moving `SwitchWithLabel` to a different PR for more feedback. This implementation works for our use-case, but we should be careful about global changes.)